dace/dtypes.py

```diff
@@ -316,6 +316,9 @@ def base_type(self):
     def veclen(self):
         return 1
 
+    def signature(self, name):
```

Comment:
Hmm, can a _type_ really have a signature? Isn't that a function term?

Reply:
I'd argue it's the signature part of that type in a function. I don't have a better name, suggestions?

Reply:
I'm unsure about this. We currently have `ctype` and will base `ocltype` that tells you how to define the type in those languages, which is already dubious to put in `dtypes` rather than in, say, `cpp.py`.
If anything, maybe `as_arg` or `as_carg`?

Reply:
@definelicht The reason why it needs a function of its own is because of C syntax unfortunately. If I want to define a callback, the callback type is not prepended before the argument name, but rather it's `return_type (*NAME)(arg1_t, arg2_t....)`. I'll change to as_arg

Reply:
Done